wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php

```diff
@@ -706,6 +707,132 @@ public function create_item( $request ) {
 	}
 
 	/**
+	 * Deletes a changeset.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		global $wpdb;
+
+		$manager = $this->ensure_customize_manager( $request['uuid'] );
+
+		if ( ! $manager->changeset_post_id() ) {
+			return new WP_Error(
+				'rest_post_invalid_uuid',
+				__( 'Invalid changeset UUID.' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		$post = get_post( $manager->changeset_post_id() );
+
+		if ( $request['force'] ) {
+			$previous = $this->prepare_item_for_response( $post, $request );
+
+			// TODO: At this point $wp_customize will no longer have up-to-date post data.
+			$result = wp_delete_post( $manager->changeset_post_id(), true );
+
+			if ( ! $result ) {
+				return new WP_Error(
+					'rest_cannot_delete',
+					__( 'The post cannot be deleted.' ),
+					array(
+						'status' => 500,
+					)
+				);
+			}
+
+			$response = new WP_REST_Response();
+
+			$response->set_data( array(
+				'deleted' => true,
+				'previous' => $previous->get_data(),
+			) );
+
+			return $response;
+		}
+
+		// TODO (?): Not filterable a la WP_REST_Posts_Controller.
+		if ( ! ( EMPTY_TRASH_DAYS > 0 ) ) {
+			return new WP_Error(
+				'rest_trash_not_supported',
+				__( 'The post does not support trashing. Set force=true to delete.' ),
+				array(
+					'status' => 501,
+				)
+			);
+		}
+
+		if ( 'trash' === get_post_status( $manager->changeset_post_id() ) ) {
```

Comment:
That's true, if revisions are enabled then a changeset keeps its `publish` status and doesn't get set to `trash`. I guess it remains to be seen if we should allow published changesets to be trashed in a UI. I suppose post revisions aren't able to be deleted in the UI? That being the case, perhaps there should be no API for trashing published changesets which are essentially revisions of an entire site's settings?